otter/convergence/service.py

```diff
@@ -0,0 +1,58 @@
+"""Converger service"""
+
+from effect import Effect
+from effect.twisted import perform
+
+from twisted.application.service import Service
+
+from otter.constants import CONVERGENCE_LOCK_PATH
+from otter.convergence.composition import (
+    execute_convergence, get_desired_group_state)
+from otter.http import TenantScope
+from otter.util.deferredutils import with_lock
+
+
+class Converger(Service, object):
+    """Converger service"""
+
+    def __init__(self, reactor, kz_client, dispatcher):
+        self._reactor = reactor
+        self._kz_client = kz_client
+        self._dispatcher = dispatcher
+
+    def _get_lock(self, group_id):
+        """Get a ZooKeeper-backed lock for converging the group."""
+        path = CONVERGENCE_LOCK_PATH.format(group_id=group_id)
+        return self._kz_client.Lock(path)
+
+    def converge(self, log, tenant_id, group_id, desired, launch_config,
+                 perform=perform,
+                 execute_convergence=execute_convergence):
+        """Converge a group to a capacity with a launch config."""
+        def exec_convergence():
+            desired_group_state = get_desired_group_state(
+                group_id, launch_config, desired)
+            eff = execute_convergence(group_id, desired_group_state)
+            eff = Effect(TenantScope(eff, tenant_id))
+            d = perform(self._dispatcher, eff)
+            return d.addErrback(log.err)
```

Comment:
Could you add `d.addErrback(log.err, "Failed to execute convergence cycle")` as it really helps when analyzing logs?